server/controllers/message.controller.js

```diff
@@ -235,4 +235,43 @@ function unarchive(req, res) {
     return res.send(req.message);
 }
 
-export default { send, reply, get, list, count, remove, load, archive, unarchive };
+/**
+ * Remove readAt timestamp
+ * sets readAt of message with messageId to null
+ * @returns {Message}
+ */
+function removeReadAt(req, res) {
+    if (req.message) {
+        req.message.update({ readAt: null });
```

Comment:
I think the `res.send` needs to be chained to the `update` promise. Otherwise the message that is returned may not be updated. Either way, the API should not return until the database operation is complete.